src/control/cmd/dmg/utils.go

```diff
@@ -41,17 +41,30 @@ func splitPort(addrPattern string, defaultPort int) (string, string, error) {
 
 	switch len(hp) {
 	case 1:
+		if defaultPort == 0 {
+			defaultPort = 10000
```

Comment:
Feels to me like the caller should be making this judgment call, since they're passing in the default.

Reply:
I tend to agree with this. It's a utility function confined to dmg so it doesn't bother me that much, but it's a little confusing. Why even accept the defaultPort parameter if it's always set to 0? Seems like it'd be better to set defaultPort as a const.

Reply:
this is a kind of awkward way of being able to pass in 0 default port if we are not concerned about the result.